### ISSUE TITLE
Pre-commit hook fails if multiple lock or requirements files are modified at same time

### ISSUE THREAD

**Issue by joakimnordling** [twu/skjold#114]
If you in your repo have multiple files that should be analyzed by skjold and use it as a pre-commit hook, the hook fails if you modify multiple lock/requirements files in the same commit.

The reason to the problem is that pre-commit tries to invoke `skjold audit` with multiple files as arguments at once and the `skjold audit` command only accepts one file at a time.

A minimal POC can be obtained by just creating an empty git repo (`mkdir skjold-poc; cd skjold-poc; git init`), enabling pre-commit (`pre-commit install`), adding a `.pre-commit-config.yaml` containing 
```
repos:
  - repo: https://github.com/twu/skjold
    rev: v0.4.1
    hooks:
    - id: skjold
      verbose: true
```
and then creating two (or more) requirements files (`touch requirements.txt requirements-dev.txt`) and then running `git add .`

If you then try to commit with `git commit -m "Adding requirements files"` you will be greeted with an error saying:
`Error: Got unexpected extra argument (requirements.txt)`. If you add even more different lock files or requirements-something.txt  files the message will be expanded with more extra arguments.

It's probably not that common people would have both poetry.lock and Pipfile.lock files, but possible people want to have multiple requirements.txt files (one for dev and one for prod) or multiple lock files in different directories of the same repo and get them all analyzed. It's of course possible to somewhat well avoid the problem by modifying only one lock file at a time, but the same issue also affects `pre-commit run --all-files` that will scan all the matched files regardless of if they've been changed or not.

**Comment by joakimnordling** [comment]
In case anyone is looking for a work around, it's possible to define the hook multiple times and each time lock it to just one file to monitor like this:
```yaml
repos:
  - repo: https://github.com/twu/skjold
    rev: v0.4.1
    hooks:
      - id: skjold
        name: "skjold-root"
        files: "^poetry.lock$"
  - repo: https://github.com/twu/skjold
    rev: v0.4.1
    hooks:
      - id: skjold
        name: "skjold-foo"
        files: "^foo/poetry.lock$"
```

**Comment by twu** [comment]
> In case anyone is looking for a work around, it's possible to define the hook multiple times and each time lock it to just one file to monitor like this:
>
> ```
> ...
> ```

Kiitos! Yes, had the same issue as you in a larger monorepo with multiple `.lock` filetypes and had to do the same workaround. But I totally agree that `SKIP=skjold pre-commit run --all-files` is not ideal 😅 I'm going to check if `nargs=-1` and assuming any passed files to be of the same `.lock`-type is as easy as it sounds :)

**Comment by joakimnordling** [comment]
Thanks, that would be awesome if we could get that kind of improvement!

**Comment by twu** [comment]
> Thanks, that would be awesome if we could get that kind of improvement!

Released `v0.5.0` which should now support passing/checking multiple files. HTH :)
